ARTstor.js

```diff
@@ -1,131 +1,899 @@
+    '3': '33',
+    '4': '34',
+    '5': '35',
+    '6': '36',
+    '7': '37',
+    '8': '38',
+    '9': '39',
+    ':': '3A',
+    ';': '3B',
+    '<': '3C',
+    '=': '3D',
+    '>': '3E',
+    '?': '3F',
+    '[': '5B',
+    // '\\' : '5C',
+    ']': '5D',
+    '^': '5E',
+    '_': '5F',
+    '`': '60',
+    '{': '7B',
+    '|': '7C',
+    '}': '7D',
+    '~': '7E'
+};
+
+/**
+    Converts two character number character to special character
+**/
+function decrypt(s) {
+    var len = s.length;
+    var i = 0;
+    var newS = '';
+    var key;
+    while (i < len) {
+        var ch = s.charAt(i);
+        if ((ch >= '0') && (ch <= '9')) {
+            key = s.slice(i, i + 2);
+            newS = newS + decodeMap[key];
+            i = i + 2;
+        } else {
+            if (ch == "!") {
+                var nCh = '';
+                if ((i + 1) < len) {
+                    nCh = s.charAt(i + 1);
+                }
+                if ((nCh >= '0') && (nCh <= '9')) {
+                    var rStr = s.substr(i + 1);
+                    var code = parseInt(rStr);
+                    nCh = String.fromCharCode(code);
+                    var nIdx = rStr.indexOf("!");
+                    if (nIdx > 0) {
+                        i = i + nIdx + 2;
+                        newS = newS + nCh;
+                    }
+                } else {
+                    newS = newS + ch;
+                    i++;
+                }
+            } else {
+                newS = newS + ch;
+                i++;
+            }
+        }
+    }
+    return newS;
+}
+
+/**
+    Converts speial character to two digit code.
+**/
+function encrypt(s) {
+    var newS = '';
+    if (s !== undefined) {
+        var len = s.length;
+        i = 0;
+        var ch;
+        while (i < len) {
+            ch = s.charAt(i);
+            if (((ch >= 'a') && (ch <= 'z')) ||
+                ((ch >= 'A') && (ch <= 'Z'))) {
+                newS = newS + ch;
+            } else {
+                var eCh = encodeMap[ch];
+                if (eCh === undefined) {
+                    newS = newS + "!" + ch.charCodeAt(0) + "!";
+                } else {
+                    newS = newS + "%" + eCh;
+                }
+            }
+            i++;
+        }
+    }
+    return newS;
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+    {
+        "type": "artwork",
+        "url": "http://www.sscommons.org/openlibrary/welcome.html#3|collections|7730455||zModeBryn20Mawr20College20Faculty2FStaff2FStudent20Photographs||||||",
+        "defer": true,
+        "items": [
+            {
+                "itemType": "artwork",
+                "title": "Trailer Home; Exterior view",
+                "creators": [
+                    {
+                        "firstName": "Barbara",
+                        "lastName": "Lane",
+                        "creatorType": "Image by"
+                    }
+                ],
+                "date": "Photographed: 2001",
```

Comment:
It's not critical and we would take this as is, but if you're able to clean up the data to be the date of creation without any additional labels, that would be ideal. Zotero is designed with the idea that Date field contains the date of creation. To obtain correct formatting in citations, the user will have to clean up the data manually otherwise.